### DISCUSSION TITLE
how to update flux images with the latest dependencies

### DISCUSSION THREAD

**Discussion by jack-evans** [fluxcd/flux2#1677] [Q&A] [answered]
Images are currently tagged and released in a new version of the flux cli.

I have found that when the images are flagged with vulnerable packages (CVEs), there is no way to remediate them unless I wait for a new release in the flux2 cli. I tried the following dockerfile:

```dockerfile
FROM ghcr.io/fluxcd/helm-controller:v0.11.1

RUN apk update && \
    apk --no-cache upgrade && \
    rm -rf /var/cache/apk/*
```

however you get permission errors which I'm guessing are to do with the user not being root when running `apk` commands.

I also tried:
```dockerfile
FROM alpine:3.14 as apkupdates

RUN apk update && \
    apk --no-cache upgrade && \
    rm -rf /var/cache/apk/*

FROM ghcr.io/fluxcd/helm-controller:v0.11.1

COPY --from=apkupdates / /
```

However I then get `Error: failed to create containerd container: mount callback failed on /var/data/cripersistentstorage/tmpmounts/containerd-mount859818087: no users found` when trying to start the pod.

Is there any documented way for a consumer to update the images themselves?

**Comment by stefanprodan** [comment, marked as answer]
I think this should work:
```dockerfile
FROM ghcr.io/fluxcd/helm-controller:v0.11.1

USER root

RUN apk update && \
    apk --no-cache upgrade && \
    rm -rf /var/cache/apk/*

USER controller

ENTRYPOINT [ "/sbin/tini", "--", "helm-controller" ]
```

**Reply by jack-evans** [reply]
ahh yes of course! that works, thanks for your help 🙂
